exec/java-exec/src/main/java/org/apache/drill/exec/store/pcap/schema/Schema.java

```diff
@@ -78,4 +81,23 @@ public ColumnDto getColumnByIndex(int i) {
   public int getNumberOfColumns() {
     return columns.size();
   }
+
+  public TupleMetadata buildSchema(SchemaBuilder builder) {
+    for(ColumnDto column : columns) {
+      if(column.getColumnType() == PcapTypes.BOOLEAN) {
+        builder.addNullable(column.getColumnName(), TypeProtos.MinorType.BIT);
+      } else if(column.getColumnType() == PcapTypes.INTEGER) {
+        builder.addNullable(column.getColumnName(), TypeProtos.MinorType.INT);
+      } else if(column.getColumnType() == PcapTypes.STRING) {
+        builder.addNullable(column.getColumnName(), TypeProtos.MinorType.VARCHAR);
+      } else if(column.getColumnType() == PcapTypes.TIMESTAMP) {
+        builder.addNullable(column.getColumnName(), TypeProtos.MinorType.TIMESTAMP);
+      } else if(column.getColumnType() == PcapTypes.LONG) {
+        builder.addNullable(column.getColumnName(), TypeProtos.MinorType.BIGINT);
```

Comment:
Nit: would be simpler to do something like the following:

```
  for (ColumnTdo column : columns) {
    builder.addNullable(column.getColumnName(), convertType(column));
  }
...
MinorType convertType(ColumnDto column) {
  switch (column.getColumnType()) {
    case PcapTypes.BOOLEAN:
      return MinorType.BIT;
   ...
```

The above requires that we handle the case of a type that is not in the switch: maybe throw an exception or some such.

Also, if the types are integers and configuous (which they are of PcapTypes is an enum), then you can create a mapping table:

```
  MinorType typeMap[] = new int[PcapTypes.getSize()];
  typeMap[PcapTypes.BOOLEAN.getOrdinal()] = MinorType.BIT;
  ...
```

Then:

```
  for (ColumnTdo column : columns) {
    builder.addNullable(column.getColumnName(), typeMap[column.getColumType().getOrdinal());
```

The mapping table is the fastest solution. But, this is not super critical in a once-per-file activity.

Reply:
I feel the need for speed... Fixed and went for the fastest solution.